internal/services/network/virtual_network_resource.go

```diff
@@ -753,9 +774,22 @@ func expandVirtualNetworkProperties(ctx context.Context, client virtualnetworks.
 				subnetObj.Properties = &virtualnetworks.SubnetPropertiesFormat{}
 			}
 
-			subnetObj.Properties.AddressPrefix = pointer.To(subnet["address_prefix"].(string))
+			if !features.FourPointOhBeta() {
+				subnetObj.Properties.AddressPrefix = pointer.To(subnet["address_prefix"].(string))
+			}
 
 			if features.FourPointOhBeta() {
+				addressPrefixes := make([]string, 0)
+				for _, prefix := range subnet["address_prefixes"].([]interface{}) {
+					addressPrefixes = append(addressPrefixes, prefix.(string))
+				}
+
+				if len(addressPrefixes) == 1 {
+					subnetObj.Properties.AddressPrefix = pointer.To(addressPrefixes[0])
+				} else {
+					subnetObj.Properties.AddressPrefixes = pointer.To(addressPrefixes)
+				}
+
```

Comment:
Do we need to `nil` out anything here to account for change from 1 prefix to 2+?

Reply:
This part is only called in the create, so there shouldn't be anything to `nil` here